open-reac/src/main/java/com/powsybl/openreac/parameters/input/OpenReacParameters.java

```diff
@@ -0,0 +1,168 @@
+
+    private final Map<String, Pair<Double, Double>> specificVoltageLimits;
+    private final List<String> variableShuntCompensators;
+    private final List<String> constantQGenerators;
+    private final List<String> variableTwoWindingsTransformers;
+    private final List<OpenReacAlgoParam> algoParamsList;
+
+    public OpenReacParameters() {
+        this.variableShuntCompensators = new ArrayList<>();
+        this.constantQGenerators = new ArrayList<>();
+        this.variableTwoWindingsTransformers = new ArrayList<>();
+        this.specificVoltageLimits = new HashMap<>();
+        this.algoParamsList = new ArrayList<>();
+    }
+
+    /**
+     * A list of shunt compensators which susceptance should be considered as variable by the optimizer.
+     * The otpimizer computes a continuous value that is rounded when results are integrated in the network.
+     */
+    public OpenReacParameters addVariableShuntCompensators(List<String> shuntsIds) {
+        this.variableShuntCompensators.addAll(shuntsIds);
+        return this;
+    }
+
+    /**
+     * Override some voltage level limits in the network. This will NOT modify the network object.
+     * @param specificVoltageLimits map containing keys : VoltageLevelId, and values are the low and high delta limits.
+     */
+    public OpenReacParameters addSpecificVoltageLimit(Map<String, Pair<Double, Double>> specificVoltageLimits) {
+        this.specificVoltageLimits.putAll(specificVoltageLimits);
+        return this;
+    }
+
+    /**
+     * A list of generators that are not controlling voltage during the optimization.
+     * The reactive power produced by the generator is constant and equals `targetQ`.
+     */
+    public OpenReacParameters addConstantQGenerators(List<String> generatorsIds) {
+        this.constantQGenerators.addAll(generatorsIds);
+        return this;
+    }
+
+    /**
+     * A list of two windings transformers which ratio should be considered as variable by the optimizer.
+     */
+    public OpenReacParameters addVariableTwoWindingsTransformers(List<String> transformerIds) {
+        this.variableTwoWindingsTransformers.addAll(transformerIds);
+        return this;
+    }
+
+    public OpenReacParameters addAlgorithmParam(List<OpenReacAlgoParam> params) {
+        this.algoParamsList.addAll(params);
+        return this;
+    }
+
+    /**
+     * Will use {@link OpenReacOptimisationObjective#BETWEEN_HIGH_AND_LOW_VOLTAGE_PROFILE} and {@link OptimisationVoltageRatio}.
+     * DO NOT CALL THIS MULTIPLE TIMES.
+     */
+    public OpenReacParameters addRatioVoltageObjective(double ratio) {
+        return addAlgorithmParam(List.of(new OptimisationVoltageRatio(ratio), OpenReacOptimisationObjective.BETWEEN_HIGH_AND_LOW_VOLTAGE_PROFILE));
+    }
+
+    public List<String> getVariableShuntCompensators() {
+        return variableShuntCompensators;
+    }
+
+    public Map<String, Pair<Double, Double>> getSpecificVoltageDelta() {
```

Comment:
Instead of a meaningless Pair<Double, Double> we could have a more business class like `VoltageLimitsShift`.
Maybe also shift would be better that delta?